src/app-admin.js

```diff
@@ -0,0 +1,491 @@
+      FROM files
+      WHERE files.id = ? and (files.owner = ?)`,
+    [fileId, userId],
+  ) || { granted: 0 };
+
+  if (granted === 0 && !isAdmin(userId)) {
+    sendErrorResponse(
+      res,
+      400,
+      'file-denied',
+      "You don't have permissions over this file",
+    );
+    return false;
+  }
+
+  const { id: fileIdInDb } = getFileById(fileId);
+  if (!fileIdInDb) {
+    sendErrorResponse(res, 400, 'invalid-file-id', 'File not found at server');
+    return false;
+  }
+
+  return true;
+}
+
+app.post('/access', (req, res) => {
+  const userAccess = req.body || {};
+  const session = validateUser(req, res);
+
+  if (!session) return;
+
+  if (!checkFilePermission(userAccess.fileId, session.user_id, res)) return;
+
+  if (!userAccess.userId) {
+    sendErrorResponse(res, 400, 'user-cant-be-empty', 'User cannot be empty');
+    return;
+  }
+
+  const { cnt } =
+    getAccountDb().first(
+      'SELECT count(*) AS cnt FROM user_access WHERE user_access.file_id = ? and user_access.user_id = ?',
+      [userAccess.fileId, userAccess.userId],
+    ) || {};
+
+  if (cnt > 0) {
+    sendErrorResponse(
+      res,
+      400,
+      'user-already-have-access',
+      'User already have access',
+    );
+    return;
+  }
+
+  getAccountDb().mutate(
+    'INSERT INTO user_access (user_id, file_id) VALUES (?, ?)',
+    [userAccess.userId, userAccess.fileId],
+  );
+
+  res.status(200).send({ status: 'ok', data: {} });
+});
+
+app.post('/access/delete-all', (req, res) => {
+  const fileId = req.query.fileId;
+  const session = validateUser(req, res);
+  if (!session) return;
+
+  if (!checkFilePermission(fileId, session.user_id, res)) return;
+
+  const ids = req.body.ids;
+  let totalDeleted = 0;
+  ids.forEach((item) => {
+    const accessDeleted = getAccountDb().mutate(
+      'DELETE FROM user_access WHERE user_id = ?',
+      [item],
+    ).changes;
+    totalDeleted += accessDeleted;
+  });
+
+  if (ids.length === totalDeleted) {
+    res
+      .status(200)
+      .send({ status: 'ok', data: { someDeletionsFailed: false } });
+  } else {
+    sendErrorResponse(res, 400, 'not-all-deleted', '');
+  }
+});
+
+app.get('/access/users', async (req, res) => {
+  const fileId = req.query.fileId;
+  const session = validateUser(req, res);
+  if (!session) return;
+
+  if (!checkFilePermission(fileId, session.user_id, res)) return;
+
+  const users = getAccountDb().all(
+    `SELECT users.id as userId, user_name as userName, display_name as displayName, 
+            CASE WHEN user_access.file_id IS NULL THEN 0 ELSE 1 END as haveAccess,
+            CASE WHEN files.id IS NULL THEN 0 ELSE 1 END as owner
+      FROM users
+      LEFT JOIN user_access ON user_access.file_id = ? and user_access.user_id = users.id
+      LEFT JOIN files ON files.id = ? and files.owner = users.id
+      WHERE users.enabled = 1 AND users.user_name <> ''`,
+    [fileId, fileId],
+  );
+  res.json(users);
+});
+
+app.post('/access/get-bulk', (req, res) => {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏what does this do? Is it used?

Reply:
not in use. removed